src/lib/parse_pdbqt.cpp

```diff
@@ -211,134 +231,130 @@ struct parsing_struct {
+            p.add(parse_pdbqt_atom_string(str), c);
+        else if(starts_with(str, "ENDROOT"))
+            return;
         else if(starts_with(str, "MODEL"))
-            throw stream_parse_error(count, "Unexpected multi-MODEL input. Use \"vina_split\" first?");
-        else throw stream_parse_error(count, "Unknown or inappropriate tag");
+            throw pdbqt_parse_error("Unexpected multi-MODEL tag found in flex residue or ligand. Use \"vina_split\" first.");
+        else 
+            throw pdbqt_parse_error("Unknown or inappropriate tag found in flex residue or ligand.", str);
     }
 }
 
-void parse_pdbqt_root(std::istream& in, unsigned& count, parsing_struct& p, context& c) {
+void parse_pdbqt_root(std::istream& in, parsing_struct& p, context& c) {
     std::string str;
+
     while(std::getline(in, str)) {
         add_context(c, str);
-        ++count;
+
         if(str.empty()) {} // ignore
         else if(starts_with(str, "WARNING")) {} // ignore - AutoDockTools bug workaround
         else if(starts_with(str, "REMARK")) {} // ignore
         else if(starts_with(str, "ROOT")) {
-            parse_pdbqt_root_aux(in, count, p, c);
+            parse_pdbqt_root_aux(in, p, c);
             break;
         }
         else if(starts_with(str, "MODEL"))
-            throw stream_parse_error(count, "Unexpected multi-MODEL input. Use \"vina_split\" first?");
-        else throw stream_parse_error(count, "Unknown or inappropriate tag");
+            throw pdbqt_parse_error("Unexpected multi-MODEL tag found in flex residue or ligand. Use \"vina_split\" first.");
+        else 
+            throw pdbqt_parse_error("Unknown or inappropriate tag found in flex residue or ligand.", str);
     }
 }
 
-void parse_pdbqt_branch(std::istream& in, unsigned& count, parsing_struct& p, context& c, unsigned from, unsigned to); // forward declaration
+void parse_pdbqt_branch(std::istream& in, parsing_struct& p, context& c, unsigned from, unsigned to); // forward declaration
 
-void parse_pdbqt_branch_aux(std::istream& in, unsigned& count, const std::string& str, parsing_struct& p, context& c) {
+void parse_pdbqt_branch_aux(std::istream& in, const std::string& str, parsing_struct& p, context& c) {
     unsigned first, second;
-    parse_two_unsigneds(str, "BRANCH", count, first, second); 
+    parse_two_unsigneds(str, "BRANCH", first, second); 
     sz i = 0;
+
     for(; i < p.atoms.size(); ++i)
         if(p.atoms[i].a.number == first) {
             p.atoms[i].ps.push_back(parsing_struct());
-            parse_pdbqt_branch(in, count, p.atoms[i].ps.back(), c, first, second);
+            parse_pdbqt_branch(in, p.atoms[i].ps.back(), c, first, second);
             break;
         }
+
     if(i == p.atoms.size())
-        throw stream_parse_error(count, "No atom number " + boost::lexical_cast<std::string>(first) + " in this branch");
+        throw pdbqt_parse_error("No atom number in this branch.", str);
```

Comment:
I think the original error message is a bit more helpful and correct - the issue isn't that there aren't any atom numbers in the branch but that a particular one couldn't be found.

Reply:
Yep, you were right!